src/runtime/actor/mod.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{error::Error, fmt::Debug};
+
+use async_trait::async_trait;
+use futures::StreamExt;
+
+use self::context::ActorContext;
+
+/// Module containing the actor context
+pub mod context;
+/// Module containing event types
+pub mod envelope;
+/// Module containing actor error types
+pub mod error;
+/// Module containing the actor address handle
+pub mod handle;
+/// Module containing actor exit report types
+pub mod report;
+
+/// The actor trait, which defines a task that is managed by the runtime
+#[async_trait]
+pub trait Actor: Send + Sync + Sized {
+    /// Custom data that is passed to all actor methods
+    type Data: Debug + Send + Sync;
```

Comment:
As an extension to my comment above: We probably still need this for everything that we initialize in `init`, or could this be removed?

Reply:
It is simply there to allow the actor a place to put data it creates without having a bunch of Options in it's own struct.